assets/js/modules/adsense/components/dashboard/AdSensePerformanceWidget.js

```diff
@@ -16,156 +16,134 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
 import { isUndefined } from 'lodash';
 
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
+
 import {
-	getTimeInSeconds,
 	readableLargeNumber,
 } from '../../../../util';
-import { TYPE_MODULES } from '../../../../components/data';
 import DataBlock from '../../../../components/data-block.js';
 import PreviewBlock from '../../../../components/preview-block';
-import { isDataZeroAdSense } from '../../util';
-import withData from '../../../../components/higherorder/withdata';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { STORE_NAME } from '../../datastore/constants';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
 
-class AdSensePerformanceWidget extends Component {
-	constructor( props ) {
-		super( props );
+const { useSelect } = Data;
 
-		this.state = {
-			twentyEightDays: false,
-			prev28Days: false,
+function AdSensePerformanceWidget( ) {
```

Comment:
Regarding my above comment, the problem _may_ only be here: This component used to rely on `withData`, which is how the `handleDataSuccess` and `handleDataError` (which are not a great approach but are what is used right now) worked before. Now that that is gone, it causes the problem I mentioned above, where the `AdSenseDashboardZeroData` component is no longer showing. Let's either reverse the refactoring or can you suggest an alternative implementation based on what you have here that works?